Description: Fix paths for eccodes, magics++ search
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2021-09-03
Forwarded: not-needed

--- a/setup.py
+++ b/setup.py
@@ -28,8 +28,8 @@
 
 assert version
 
-libdir = os.path.realpath("install/lib")
-incdir = os.path.realpath("install/include")
+libdir = os.path.realpath("/usr/lib")
+incdir = os.path.realpath("/usr/include")
 libs = ["eccodes", "MagPlus"]
 
 
